src/main/java/teammates/storage/sqlentity/Notification.java

```diff
@@ -0,0 +1,301 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.SanitizationHelper;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+/**
+ * Represents a unique notification in the system.
+ */
+@Entity
+@Table(name = "Notifications")
+public class Notification extends BaseEntity {
+
+    @Id
+    private String notificationId;
+
+    @Column(nullable = false)
+    private Instant startTime;
+
+    @Column(nullable = false)
+    private Instant endTime;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.ORDINAL)
```

Comment:
Any opinions for choosing `EnumType.ORDINAL` over `EnumType.STRING`?

Reply:
`EnumType.ORDINAL` is likely to be more efficient, but it has some downsides: less readable in the database and we cannot reorder the enum definition. I believe it depends on our use cases. What is your advice?

Reply:
I would prefer `EnumType.STRING`, reason being that it will be easier to maintain and less prone to bugs, with except maybe a slight penalty on performance. For `EnumType.ORDINAL`, if we add a new value in the middle or rearrange the enum's order, we'll break the existing data model. Such issues might be hard to catch as well as problematic to fix since we would have to update all the database records.

Reply:
Ok! Will update!

Reply:
Agree with the use of strings!